pkg/pub_integration/lib/src/fake_test_user.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:_pub_shared/pubapi.dart';
+import 'package:http/http.dart' as http;
+import 'package:pub_integration/src/fake_credentials.dart';
+import 'package:pub_integration/src/headless_env.dart';
+import 'package:pub_integration/src/pub_puppeteer_helpers.dart';
+import 'package:puppeteer/puppeteer.dart';
+
+import 'test_scenario.dart';
+
+class FakeTestUser implements TestUser {
```

Comment:
Let's make this private. I don't see any reason why anyone should ever do `user is FakeTestUser`, is there a reason to do this?